UnitTests/GitCommands.Tests/Submodules/SubmoduleStatusProviderTests.cs

```diff
@@ -130,14 +150,10 @@ public async Task Submodule_status_changes_for_top_module_with_first_nested_modu
 
             // Revert the change
             File.Delete(Path.Combine(_repo2Module.WorkingDir, "test.txt"));
-            changedFiles = GetStatusChangedFiles(currentModule);
-            changedFiles.Should().HaveCount(0);
-            await SubmoduleTestHelpers.UpdateSubmoduleStatusAndWaitForResultAsync(_provider, currentModule, changedFiles);
-            result.AllSubmodules.All(i => i.Detailed == null).Should().BeTrue();
-            result.OurSubmodules.Should().BeEquivalentTo(result.AllSubmodules);
```

Comment:
Why has this been removed nearly all testcases?

Reply:
Because CheckRevertedStatus() is called
There is no check that the cleanup was partially done as it was before

Reply:
I mean just the line `result.OurSubmodules.Should().BeEquivalentTo(result.AllSubmodules);` which is not part of `CheckRevertedStatus`.

Reply:
See below, tested separately